### PR TITLE
Extend torch.utils._sympy.symbol for more Inductor symbols

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #125496
* #125494
* #125483
* __->__ #125419

I'm still missing a few, cdzq at least

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -1569,7 +1571,7 @@ def split_and_set_ranges(self, lengths: List[List[sympy.Expr]]):
 
     def is_indirect_indexing(self, index: sympy.Expr):
```
Comment (lezcano) [inline torch/_inductor/codegen/triton.py:1572]: Do we want to reuse this function everywhere? It'd be quite nice.

File: torch/_inductor/ir.py [hunk 2]
```diff
@@ -2212,7 +2216,9 @@ def _dynamic_reshape_indexer(old_size, new_size):
         Perform a reshape entirely by modifying indexing math
         """
         size_hint = V.graph.sizevars.size_hint
-        vars = [sympy_index_symbol(f"view{i}") for i in range(len(new_size))]
+        vars = [
+            sympy_index_symbol_with_prefix(SymT.VIEW, i) for i in range(len(new_size))
```
Comment (lezcano) [inline torch/_inductor/ir.py:2220]: does this ever leave this function? (first time I see this symbol).
If not, we could potentially have a `SymT.AUX` to cover all these cases where you want a dummy variably.

File: torch/_inductor/ir.py [hunk 3]
```diff
@@ -4410,6 +4416,7 @@ def canonicalize(self):
         sizes = self.get_size()
         strides = self.get_stride()
         strides = [sizevars.size_hint(x) for x in strides]
+        # TODO: I can't tell if the symbols here are temporary
```
Comment (lezcano) [inline torch/_inductor/ir.py:4419]: assert that they are not part of `index` at the end of the function?
They look temporary to me, as they are all replaced in the `sympy_subs` below. The interesting thing is that they are replaced by a new variable `ci` which does escape this function...

File: torch/_inductor/ir.py [hunk 4]
```diff
@@ -7744,8 +7751,7 @@ def add_submodule(self, block, prefix):
         return name
 
     def add_indirect(self, size):
-        name = f"indirect{len(self.indirect_vars)}"
-        var = sympy_index_symbol(name)
+        var = sympy_index_symbol_with_prefix(SymT.INDIRECT, len(self.indirect_vars))
```
Comment (lezcano) [inline torch/_inductor/ir.py:7754]: This is a placeholder variable that is later replaced with `SymT.TMP`. Can you add this to the description of the `SymT` symbol?

**Review by lezcano — APPROVED** [review]
Love this effort! Left a few comments, but otherwise LGTM

File: torch/_inductor/codegen/triton.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/codegen/triton.py:1572]: I'm not... opposed? But it also doesn't seem like that much benefit haha

File: torch/_inductor/ir.py (hunk 2 above)
Comment (ezyang) [reply torch/_inductor/ir.py:2220]: I don't know. There are a few other variables where I don't know if they leave. If I care enough (but I think I am at the point of not caring), I will add asserts to see if they escape or not.

File: torch/_inductor/ir.py (hunk 3 above)
Comment (ezyang) [reply torch/_inductor/ir.py:4419]: I'm going to do the asserts separately as they're likely to fail

File: torch/_inductor/codegen/triton.py (hunk 1 above)
Comment (lezcano) [reply torch/_inductor/codegen/triton.py:1572]: It's just easier to understand what's going on. `TMP` has never been a particularly descriptive name lol

**Pull request closed without merging**